src/event-input/calcPb.js

```diff
@@ -13,26 +16,25 @@ const moment = require('moment');
  * @returns { { result: object, runner: object } } The modified result and runner objects
  */
 const calcPb = (result, runner) => {
-    let newPb = false;
+    let newPb = 0;
     
-    let distance = result.distance;
+    let distance = result.distance.toString();
     if (!['2', '5'].includes(distance)) throw new Error('distance invalid');
 
     let fastest = runner.stats[`records${distance}k`].fastest;
     let slowest = runner.stats[`records${distance}k`].slowest;
 
 
-    if (!fastest) {
+    if (!fastest /* Therefore also !slowest */) {
         runner.stats[`records${distance}k`].fastest = result;
-    } else if (!slowest) {
-        runner.stats[`records${distance}k`].slowest = result;
+        runner.stats[`records${distance}k`].slowest = result;        
```

Comment:
We should start moving towards "functional" programming whereby we do not alter any of the parameters in the function. Altering the parameters changes the variables outside of the function's scope and can have unintended side-effects. I know that you pass back the changed runner at the end of this function, but it's a good practice to adopt whereby you do not modify the parameter, just pass back a new version of the runner. You can do this either by cloning the data at the top of the function, or when you pass it back, pass back a merge of the old and the new e.g.:

```js
return {
	runner: {
		...runnerFromParameter,
		stats: newStats
	}
};
```